qiskit/compiler/assemble.py

```diff
@@ -219,6 +220,9 @@ def _parse_common_args(backend, qobj_id, qobj_header, shots,
             shots = min(1024, max_shots)
         else:
             shots = 1024
+    elif not isinstance(shots, int):
+        raise TypeError(
+            "Argument 'shots' should be of type 'int'")
```

Comment:
```suggestion
    elif not isinstance(shots, int):
        raise QiskitError(
            "Argument 'shots' should be of type 'int'"
     )
```

Reply:
Could you just clarify whether I should raise TypeError or QiskitError?

Reply:
@taalexander suggests `QiskitError`.

Reply:
Ohk thanks, I got it.